Issue forms/templates

benjyw opened the pull request:
For:

  - Bug reports
  - Feature requests
  - Adding a company to a public list of users

The driving goal was the user registration template, but it made it a bit weird, when opening a new issue, to have a template for that and not for bugs and feature requests. The user would have to click on a small "open a blank issue" link to actually file a normal issue. So this change also includes the bug and feature templates. Those are purposely simple though, and based on GitHub's suggested templates. No need to make bug/feature reporters jump through form hoops.

[ci skip-rust]

[ci skip-build-wheels]

benjyw commented:
Template chooser renders as:

<img width="1183" alt="Screen Shot 2021-08-20 at 3 50 53 PM" src="https://user-images.githubusercontent.com/512764/130286559-27c16488-40e3-420e-baeb-79b48a83e246.png">

benjyw commented:
User registration form renders as:

<img width="968" alt="Screen Shot 2021-08-20 at 3 52 28 PM" src="https://user-images.githubusercontent.com/512764/130286685-41b3cc2d-0610-4a67-a03d-ea4353ea8970.png">

benjyw commented:
Bug report renders as:

<img width="991" alt="Screen Shot 2021-08-20 at 3 53 13 PM" src="https://user-images.githubusercontent.com/512764/130286763-9e6c842a-47bb-4d60-80c3-69e3f0729030.png">

Eric-Arellano commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,20 @@
+---
+name: Bug report
+about: Help us make Pants better
```
This sounds very similar to feature request. Maybe "more reliable"? Or "Help us squash bugs"?

Eric-Arellano commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,20 @@
+Are you encountering the bug on MacOS, Linux, or both?
+
+**Additional info**
+Add any other information about the problem here, including attachments or links to gists.
```
This makes it sound like you must have a gist.

```suggestion
Add any other information about the problem here, e.g. any attachments or links to gists.

It's often helpful to run with `./pants -ldebug` and copy and paste all the output into a gist.
```

benjyw commented:
Feature request renders as:

<img width="1013" alt="Screen Shot 2021-08-20 at 3 53 59 PM" src="https://user-images.githubusercontent.com/512764/130286820-1f62839d-4093-4f4e-8cc6-363748fba931.png">

Eric-Arellano commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,20 @@
+---
+
+**Is your feature request related to a problem? Please describe.**
+A clear and concise description of what the problem is. E.g., I'm always frustrated when [...]
```
Let's frame it in a more positive light, especially because we can't guarantee we'll implement their request. It's about making their experience even better, rather than their experience being bad to begin with.

```suggestion
A clear and concise description of what the problem is. E.g., My life would be better if [...]
```

Eric-Arellano reviewed (approved):
Awesome!

cczona commented:
For the user registration form, I suggest emphasizing that the email address is not for publication on the website. Also, "registration" typically refers to things like registering for warranty support or other customer support. So I wonder if that title would be dismissed as "I don't need this", rather than "Sure, I'll help support Pants by doing this". What if the title is something like "Support the project: add your company to user.pantsbuild.org"?

cczona commented:
I like that you included the docs and Slack. Might as well also add the blog, twitter feed, and YouTube channel.

https://www.youtube.com/channel/UCCcfCbDqtqlCkFEuENsHlbQ

benjyw commented:
> I like that you included the docs and Slack. Might as well also add the blog, twitter feed, and YouTube channel.
> 
> https://www.youtube.com/channel/UCCcfCbDqtqlCkFEuENsHlbQ

I'd rather not, because those aren't really places for getting help, which is what the focus is on here. And since the "template chooser" is already quite busy, I think we should keep it to just the highest value things.

benjyw commented:
> For the user registration form, I suggest emphasizing that the email address is not for publication on the website. Also, "registration" typically refers to things like registering for warranty support or other customer support. So I wonder if that title would be dismissed as "I don't need this", rather than "Sure, I'll help support Pants by doing this". What if the title is something like "Support the project: add your company to user.pantsbuild.org"?

Changed the text relating to the email address to emphasize this. 

Changed the title to "Support Pants - List My Company on pantsbuild.org". The description and other text already references the Pants Users list.

stuhood commented on .github/ISSUE_TEMPLATE/user_registration.yml:
```diff
@@ -0,0 +1,54 @@
+      placeholder: e.g., https://www.acme.com
+    validations:
+      required: true
+  - type: input
+    id: email
+    attributes:
+      label: Contact email
+      description: |
+        How can we contact you if we have questions about this form? We *won't* publish your
+        email address on the Pants Users list, or use it for any other purpose.
+      placeholder: e.g., wilecoyote@acme.com
+    validations:
+      required: true
```
I don't think that we strictly need to ask for an email address here, given that we can follow up on the created github issue to ask further questions. I expect that a non-trivial number of users will be uncomfortable with (or perhaps surprised by) having their email address show up in the issue.

benjyw replied:
Fixed.

stuhood commented on .github/ISSUE_TEMPLATE/user_registration.yml:
```diff
@@ -0,0 +1,54 @@
+
+        If you're unsure, please submit this page without the logo, and revisit it later, when
+        permission has been established. The other information is still very helpful!
+      placeholder: Drag and drop a .jpg or .png image of your company logo into this box .
```
```suggestion
      placeholder: Drag and drop a .jpg or .png image of your company logo into this box.
```

benjyw replied:
Fixed.

stuhood reviewed (approved):
Thanks!

benjyw commented:
User form now renders as:

<img width="664" alt="Screen Shot 2021-08-23 at 1 10 38 PM" src="https://user-images.githubusercontent.com/512764/130512880-db9f93dd-d819-4550-a929-6b3a09b6e5b3.png">

cczona commented:
For the user registration form, the main purpose is to validate that the submitter is affiliated with the organization they're adding. Contacting them with questions can typically be done via the issue itself, and in the rare private case taken to DMs or email. (We don't care what address someones uses when discussing questions. Whereas, ensuring that the Who Uses Pants list is accurate is important.)

I propose that instead of asking for email address, we treat the form as Part 1 of 2, with Part 2 being that they send an email (empty is fine) with subject "verify" from their company's domain to an address at pantsbuild.org (verify@pantsbuild.org?) which would be monitored by several Maintainers via Google Groups. The issue would only be approved after a Maintainer has established that the domain for the new listing and the submitter match. Then the Google Group would be available as audit trail if anyone questions the accuracy of a listing.

Eric-Arellano commented:
That's similar to how PyPI was doing two-factor authentication restores: you'd email each other before the issue could be closed. It worked well!

benjyw commented:
> For the user registration form, the main purpose is to validate that the submitter is affiliated with the organization they're adding. Contacting them with questions can typically be done via the issue itself, and in the rare private case taken to DMs or email. (We don't care what address someones uses when discussing questions. Whereas, ensuring that the Who Uses Pants list is accurate is important.)
> 
> I propose that instead of asking for email address, we treat the form as Part 1 of 2, with Part 2 being that they send an email (empty is fine) with subject "verify" from their company's domain to an address at pantsbuild.org ([verify@pantsbuild.org](mailto:verify@pantsbuild.org)?) which would be monitored by several Maintainers via Google Groups. The issue would only be approved after a Maintainer has established that the domain for the new listing and the submitter match. Then the Google Group would be available as audit trail if anyone questions the accuracy of a listing.

Yep, see screenshot.
